MPSGameLift/Code/Source/MPSGameLiftClientSystemComponent.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ * 
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "MPSGameLiftClientSystemComponent.h"
+
+#include <AzCore/Serialization/SerializeContext.h>
+#include <Request/AWSGameLiftRequestBus.h>
+#include <Request/AWSGameLiftSessionRequestBus.h>
+#include <Request/AWSGameLiftJoinSessionRequest.h>
+
+namespace MPSGameLift
+{
+    void MPSGameLiftClientSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<MPSGameLiftClientSystemComponent, AZ::Component>()
+                ->Version(0)
+                ;
+        }
+    }
+
+    void MPSGameLiftClientSystemComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("MPSGameLiftClientService"));
+    }
+
+    void MPSGameLiftClientSystemComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC_CE("MPSGameLiftClientService"));
+    }
+
+    void MPSGameLiftClientSystemComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("AWSGameLiftClientService"));
+    }
+
+    void MPSGameLiftClientSystemComponent::GetDependentServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& dependent)
+    {
+    }
+
+    void MPSGameLiftClientSystemComponent::Init()
+    {
+        m_playerId = AZ::Uuid::Create().ToString<AZStd::string>();
```

Comment:
It may be more flexible to store the m_playerId as a Uuid, and then convert when JoinSessionInternal is invoked.

It would allow the m_playerId to keep to constraints of the Uuid class such as comparing two Uuids

Reply:
Good call fixed